src/main/java/com/google/sps/data/UserDatastoreRepository.java

```diff
@@ -0,0 +1,130 @@
+package com.google.sps.data;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.User;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+
+/**
+ * This class implements the match repository using the datastore 
+ */
+public class UserDatastoreRepository implements UserRepository {
+    
+  private Collection<User> allUsers = new ArrayList<>();
```

Comment:
could you explain the idea behind storing a collection of Users in allUsers?

Reply:
The idea behind storing a collection of allUsers is to use this later on for storing profiles that get fetched from the database mainly.

Reply:
Could you clarify what the advantages / disadvantages of storing profiles that get fetched from the database? (i.e. why do we need to store them in an ArrayList in addition to storing them in the database?)

Reply:
## Advantages
Have an list to collect the profiles from database, makes conditional checks precise since it would check to allUsers which fetches the profiles.
## Disadvantages 
Uses space

We need to store them in the arraylist so we can track what is in it for the conditional checks

Reply:
The reason behind the Collection is to copy the datastore User profiles and then store them locally for testing , calling methods, and sending to frontend

Datastore variable is to connect with database with methods to put, delete, fetch and more.